src/lsp/cobol_cfg/cfg_builder.mli

```diff
@@ -0,0 +1,149 @@
+(******************************************************************************)
+(*                                                                            *)
+(*     Copyright (c) 2021-2023 OCamlPro SAS                                   *)
+(*                                                                            *)
+(*     All rights reserved.                                                   *)
+(*     This file is distributed under the terms of the                        *)
+(*     OCAMLPRO-NON-COMMERCIAL license.                                       *)
+(*                                                                            *)
+(******************************************************************************)
+
+open Cfg_jumps
+
+type node_type =
+  | External of string
+  | Entry of [`Point | `Paragraph | `Section of string | `Statement of string]
+  | Normal of string * string (* fullname * display_name *)
+  | Collapsed of string Cobol_common.Basics.NEL.t
+  | Split of string
+
+type node = {
+  id: int;
+  section_name: string;
+  loc: Cobol_common.srcloc option;
+  typ: node_type;
+  jumps: Jumps.t;
+  will_fallthru: bool;
+  terminal: bool; (* unused atm *)
+}
+
+module Node: sig
+  type t = node
+
+  val compare : node -> node -> int
+  val hash : node -> int
+  val equal : node -> node -> bool
+end
+
+type edge =
+  | FallThrough
+  | Perform
+  | Go
+
+module Edge: sig
+  type t = edge
+
+  val compare : 'a -> 'a -> int
+  val default : edge
+end
+
+module Cfg :
```

Comment:
Here you probably just need to re-use the `Graph.Sig.P` module type. That would be something like
```ocaml
module Cfg: Graph.Sig.P
  with type V.t = Node.t
   and type V.label = Node.t
   and type E.t = Node.t * Edge.t * Node.t
   and type E.label = Edge.t
```

Reply:
The declarations (modules `Node`, `Edge` and `Cfg`, along with types `node_type`, `node` and `edge` should be in a separate module (I suggest `Cfg_types`, bound as `Cobol_cfg.Types`).